### PR TITLE
azurerm_synapse_workspace_key - can not be correctly rotated

### PR DESCRIPTION
**Pull request by ms-henglu**
image

### CODE REVIEW

File: internal/services/synapse/synapse_workspace_key_resource_test.go [hunk 1]
```diff
@@ -15,13 +15,37 @@ import (
...
 
 	data.ResourceTest(t, r, []acceptance.TestStep{
 		{
-			Config: r.customerManagedKey(data),
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		// CMK takes a while to activate, so validation against the plan tends to fail.
```
Comment (katbyte) [inline internal/services/synapse/synapse_workspace_key_resource_test.go:29]: we should be waiting on activation then? with a state wait until it is done so this never fails?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/synapse/synapse_workspace_key_resource_test.go (hunk 1 above)
Comment (ms-henglu) [reply internal/services/synapse/synapse_workspace_key_resource_test.go]: Hi @katbyte ,

Yes, I've added the waiting part in the first commit. And I've removed this comments in the second commit. Please take another look, thanks!

**Comment by ms-henglu** [comment]
Hi @katbyte , testcases are all passed, would you please take another look? Thanks!
image

File: internal/services/synapse/synapse_workspace_key_resource_test.go (hunk 1 above)
Comment (katbyte) [reply internal/services/synapse/synapse_workspace_key_resource_test.go]: only the the update function? what about create? shouldn't it be there too?

Comment (ms-henglu) [reply internal/services/synapse/synapse_workspace_key_resource_test.go]: Yes, you're right, I've added another commit to add it in both create and update.

**Review by katbyte — APPROVED** [review]
Thanks @ms-henglu - LGTM 🏗️

**Pull request merged**